### DISCUSSION TITLE
Make rate adjustment setting on daycore and nightcore mod adjust pitch, not just rate

### DISCUSSION THREAD

**Discussion by WRMSRwasTaken** [ppy/osu#13125] [Ideas] [answered]
**Describe the new feature:** 
When enabling the Daycore or Nightcore mod and adjusting the speed decrese / increase with the Difficulty Adjust mod, it would be great in my opinion to change the pitch instead of just adjusting the song's speed with a fixed lower/higher frequency which sounds a bit weird in my opinion. Half time and double time already have a fixed frequency with different speeds.

This is already the case with the Wind Down or Wind Up mod, where changing the Final rate on both mods changes the pitch of the song.

**Proposal designs of the feature:** 
Just a code change, no new design elements needed.

**Comment by peppy** [comment]
The current implementation was done so with intention. It's really a matter of opinion as to which is more correct. Will leave this open for future discussion.

**Reply by bdach** [reply]
This was brought up to me again today by @Walavouchey - could you maybe elaborate on what was the intention behind having nightcore/daycore have a fixed 1.5x/0.75x pitch? Because to me it feels more natural to use the speed value chosen by the user directly, but I'm not sure what I may be missing there.

**Reply by peppy** [reply]
The values were chosen to match the genres in each case. Adjusting the pitch further would no longer achieve the goal of the mod.

**Reply by Sanch-KK** [reply]
Many people have found another use and purpose for this mod which doesn't break anything in any way and just makes the game more enjoyable for said people. For now to achieve the desired result you have to run stable with external programs which is kinda annoying, and if i recall lazer's motto correctly - bring in more features until you start to want to use lazer instead of stable it should be ok to accept this change

**Reply by salami0133** [reply]
> The values were chosen to match the genres in each case. Adjusting the pitch further would no longer achieve the goal of the mod.

I know this is old but wouldn't it be better to merge DT and NC and just make a pitch toggle so the rate of the song also affects the pitch?
It would sound more natural and I think most if not all nightcore enjoyers agree. 
I get that you want the mod to be authentic but no one who made a nightcore edit ever (except the "original" musical duo) has their pitch and speed at exactly 135% and F#.
The way it is now just sounds weird and kinda defeats the purpose of being able to increase the rate... it's just DT again.
Sorry for my rambling I hope you have a good day.

**Comment by WRMSRwasTaken** [comment]
Looking at the latest lazer release (or 1-2 before that) a checkbox "Adjust pitch" has been added to the Wind up and Wind down mods, maybe could be added to daycore & nightcore mods aswell (unchecked by default)?

**Comment by prismotizm** [comment]
Artificially stretching audio also causes distortion & major loss in quality so having this as an option would be amazing, especially for people who would use Nightcore instead of DoubleTime on stable just because the audio sounded cleaner.

**Comment by Parakosmos** [comment]
Just wanting to address my points out from my post, #17883, into this discussion:

- **NC and DC should stay as standalone modification**s that have the 1.5x/0.75x tempo-pitch adjustment to preserve their thematic distinction from the DT and HT mods & maintain the authenticity of replays already set with NC.

- The **option for having the beatmap's pitch to be proportional to rate adjustment should be given to the DT/HT** _for the above dotpoint's matter_ as they simply either speed up or slow down the beatmap. This kind of setting exists in Etterna as a toggable option. 

- Expanding on the above point, to further distinguish NC from DT - if let's say, the propotional pitch-tempo setting existed and a DT map was played to the same as NC at 1.5x rate - you could **force the nightcore instruments to be included when playing the NC mod itself whilst leaving them absent for a DT play played on the same beatmap (likewise with DC from HT).** 

- The section dot point from the discussion, #13142, mentioned how a **custom pitch setting** could be implemented to make the tempo-pitch adjustment difference as a freeform feature. For example a player could change the pitch of a 1.5x DT beatmap to be adjusted to 0.75x to make the audio sound deeper than intended, ending up creating an inverse effect on the tempo-pitch adjustment rather than a direct proportion effect as seen with NC and DC.

**Comment by Terratype** [comment]
I very much second having something like a button on nc/dc for proportional pitch, i play them instead of dt/ht so that theres no audio stretching/artifacting but the slider options simply ruin it by bringing it back in

**Comment by tezvii** [comment]
I also back this idea as a toggleable option in either NC/DC or DT/HT to get rid of all of the artifacting that comes with pitch-preserving timestretched audio. It doesn't need to be intended for rank either - just having the option for fun would be more than enough. McOsu already has a global console toggle for disabling the pitch preservation, so I would like to see this be added in some form, at least.

**Comment by Sanch-KK** [comment]
Strongly agree on proportional pitch change, many people including myself are using NC to avoid audio artifacts which come with speed changes without binding to pitch

**Reply by prismotizm** [reply]
Actual gameplay problem here because the audio stretching artifacts really mess up the rhythm of the song and most of the time makes it harder to play if not unplayable. I rely on the rhythm of the song as a reference for my tapping, I would imagine most other players do. I cannot just recreate the original rhythm in my head over the ruined audio to tap consistently. If I want to play a map faster I ALWAYS pick NC instead of DT, because it uses resampling instead of grain stretching (the non-pitch-changing option), and the rhythm is preserved. But I want to be able to customize the speed sometimes, and I can't do it comfortably because the "speed multiplier" value just adds grain stretching to the already 1.5x resampled "nightcore" audio, which I see as incredibly redundant, and the grain stretching artifacts end up ruining the audio again, as if I am just using DT.

**Reply by rszyma** [reply]
NC uses grain stretching too. Try turning on NC 1.01x and see yourself.

**Reply by DeltaRMR** [reply]
I agree the audio is messed up, need to make it to do resampling while changing bpm, this doesn't sound difficult and sound would be much better. Maybe instead of `multiplier` you could change it to like `BPM`.
There is some resampling methods that you can find on Wikipedia and add that feature

**Comment by Rekunan** [comment, marked as answer]
Someone has made a PR that goes about this in a different way #24640

**Reply by frenzibyte** [reply]
Marked as answer.
